velox/common/memory/SharedArbitrator.h

```diff
@@ -199,8 +211,30 @@ class SharedArbitrator : public memory::MemoryArbitrator {
   void incrementGlobalArbitrationCount();
   void incrementLocalArbitrationCount();
 
+  // Returns the max reclaimable capacity from 'pool' which includes both used
+  // and free capacities.
+  int64_t reclaimableCapacity(const MemoryPool& pool) const;
+
+  // Returns the free memory capacity that can be reclaimed from 'pool' by
+  // shrink.
+  int64_t reclaimableFreeCapacity(const MemoryPool& pool) const;
+
+  // Returns the used memory capacity that can be reclaimed from 'pool' by disk
+  // spill.
+  int64_t reclaimableUsedCapacity(const MemoryPool& pool) const;
+
+  // Returns the minimal amount of memory capacity to grow for 'pool' to have
+  // the reserved capacity as specified by 'memoryPoolReservedCapacity_'.
+  int64_t minGrowCapacity(const MemoryPool& pool) const;
```

Comment:
Is it so that memory pool reserved capacity is only applied if this pool goes under arbitration process as either a memory giver or taker?

Reply:
Explained offline.